File: Examples/Lorem Ipsum iOS Example/ViewController.m
/*
 *        __                                 ____
 *       / /   ____  ________  ____ ___     /  _/___  _______  ______ ___
 *      / /   / __ \/ ___/ _ \/ __ `__ \    / // __ \/ ___/ / / / __ `__ \
 *     / /___/ /_/ / /  /  __/ / / / / /  _/ // /_/ (__  ) /_/ / / / / / /
 *    /_____/\____/_/   \___/_/ /_/ /_/  /___/ .___/____/\__,_/_/ /_/ /_/
 *                                          /_/
 *
 *                              ViewController.m
 *                   http://github.com/lukaskubanek/LoremIpsum
 *             2013-2020 (c) Lukas Kubanek (http://lukaskubanek.com)
 */

#import "ViewController.h"
#import "LoremIpsum.h"

@interface NSArray (LoremIpsum)

- (id)li_randomObject;

@end

@interface ViewController ()

@property (weak) IBOutlet UIButton *button;
@property (weak) IBOutlet UIActivityIndicatorView *activityIndicator;
@property (weak) IBOutlet UIImageView *imageView;
@property (weak) IBOutlet UILabel *informationLabel;

- (IBAction)loadImage:(id)sender;

@end

@implementation ViewController

- (IBAction)loadImage:(id)sender
{
    self.activityIndicator.hidden = NO;
    self.button.enabled = NO;
    self.imageView.image = nil;
    self.informationLabel.text = @"Loading…";

    [self.button setTitle:@"Load Another Image" forState:UIControlStateNormal];

    NSArray *services = @[@(LIPlaceholderImageServiceLoremPixel),
            @(LIPlaceholderImageServiceDummyImage),
            @(LIPlaceholderImageServicePlaceKitten)];
    LIPlaceholderImageService service = (LIPlaceholderImageService)[[services li_randomObject] intValue];
    CGSize size = CGSizeMake(MAX(100, arc4random() % (NSInteger)self.imageView.frame.size.width),
            MAX(100, arc4random() % (NSInteger)self.imageView.frame.size.height));

    NSString *serviceString = nil;
    if (service == LIPlaceholderImageServiceLoremPixel) {
        serviceString = @"lorempixel.com";
    } else if (service == LIPlaceholderImageServiceDummyImage) {
        serviceString = @"dummyimage.com";
    } else if (service == LIPlaceholderImageServicePlaceKitten) {
        serviceString = @"placekitten.com";
    }

    NSString *information = [NSString stringWithFormat:@"%@ %0.fx%0.f", serviceString, size.width, size.height];

    [LoremIpsum asyncPlaceholderImageFromService:service
                                        withSize:size
                                      completion:^(UIImage *image) {
        self.imageView.image = image;
        self.informationLabel.text = information;

        self.activityIndicator.hidden = YES;
        self.button.enabled = YES;
    }];
}

@end

File: Examples/Lorem Ipsum tvOS Example/Lorem Ipsum tvOS Example/ViewController.swift
/*
*        __                                 ____
*       / /   ____  ________  ____ ___     /  _/___  _______  ______ ___
*      / /   / __ \/ ___/ _ \/ __ `__ \    / // __ \/ ___/ / / / __ `__ \
*     / /___/ /_/ / /  /  __/ / / / / /  _/ // /_/ (__  ) /_/ / / / / / /
*    /_____/\____/_/   \___/_/ /_/ /_/  /___/ .___/____/\__,_/_/ /_/ /_/
*                                          /_/
*
*                             ViewController.swift
*                   http://github.com/lukaskubanek/LoremIpsum
*             2013-2020 (c) Lukas Kubanek (http://lukaskubanek.com)
*/

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var imageView: UIImageView!
    @IBOutlet weak var informationLabel: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()

        loadImage()
    }

    @IBAction func userPressedReloadButton() {
        loadImage()
    }

    func loadImage() {
        let services: [LIPlaceholderImageService] = [.dummyImage,
                                                     .placeKitten]

        guard let service = services.randomElement() else { fatalError() }

        var imageSize = imageView.bounds.size

        var serviceString: String

        switch service {
        case .dummyImage: serviceString = "dummyimage.com"
        case .loremPixel: serviceString = "lorempixel.com"
        case .placeKitten: serviceString = "placekitten.com"
        }

        serviceString.append(" \(Int(imageSize.width)) x \(Int(imageSize.height))")

        informationLabel.text = "Loading…"
        imageView.image = nil

        let scale = UIScreen.main.scale
        imageSize = imageSize * scale

        LoremIpsum.asyncPlaceholderImage(from: service, with: imageSize) { (image) in
            self.imageView.image = image
            self.informationLabel.text = serviceString
        }
    }

}

func *(lhs: CGSize, rhs: CGFloat) -> CGSize {
    return CGSize(width: lhs.width * rhs, height: lhs.height * rhs)
}

